### ISSUE TITLE
Game crash when frame chests interact with xnet

### ISSUE THREAD

**Issue by ylehir** [PianoManu/BlockCarpentry#21]
The crash happens when you connect a xnet connector to a frame chest and you try to apply a texture to the frame.
See attached log.

Version 0.3.4
XNet version 3.0.8
Forge version 34.1.42

[debug.log](https://github.com/PianoManu/BlockCarpentry/files/5510878/debug.log)

**Comment by PianoManu** [comment]
I'm not sure, but I think this problem comes from XNet. My chest Tile Entity does extend the vanilla chest Tile Entity (because I need to change some things), and XNet tries to cast my Tile Entity to a connector Tile Entity (which is not possible). I don't think there's much I can do, sorry...

**Comment by PianoManu** [comment]
Does this issue happen with XNet version 3.0.11? Because what I can read from the debug log and from McJty code for xnet, it seems like he already fixed it. Could you please tell me, whether this problem still occurs?

**Comment by r-mc2** [comment]
Apologies for digging up this old (and closed) issue, but I was referred here from the BotanyPots dev as my current error seems to relate. [The related issue](https://github.com/Darkhax-Minecraft/BotanyPots/issues/138) and associated logs can be found over there.

Perhaps this will shed some more light on this original problem?

**Comment by PrincessRTFM** [comment]
I just got the same crash you reported on BotanyPots - see below - and it's definitely an issue on Block Carpentry. I'd try to track down the cause, but for some reason `ChestFrameTileEntity.java` isn't on the repo? Curiously, according to the github commit history filtered on that directory, there's actually _never been_ such a file...

All I can tell from the stack trace is that your `getCapability()` method is trying to do something with `null` and it's throwing.

<details>
<summary>Details</summary>

```
java.lang.NullPointerException: null
	at mod.pianomanu.blockcarpentry.tileentity.ChestFrameTileEntity.getCapability(ChestFrameTileEntity.java:234) ~[blockcarpentry:1.16-0.3.5a] {re:classloading}
	at me.desht.modularrouters.util.InventoryUtils.getInventory(InventoryUtils.java:52) ~[modularrouters:1.16.4-7.3.1-50] {re:classloading}
	at me.desht.modularrouters.item.module.TargetedModule.isValidTarget(TargetedModule.java:69) ~[modularrouters:1.16.4-7.3.1-50] {re:classloading}
	at me.desht.modularrouters.item.module.TargetedModule.func_195939_a(TargetedModule.java:53) ~[modularrouters:1.16.4-7.3.1-50] {re:classloading}
	at net.minecraft.item.ItemStack.lambda$onItemUse$5(ItemStack.java:185) ~[?:?] {re:mixin,re:classloading,xf:fml:forge:filled_map.4,pl:mixin:APP:performant.mixins.json:item.ItemStackMixin,pl:mixin:APP:dankstorage.mixins.json:ItemStackMixin,pl:mixin:APP:quark.mixins.json:ItemStackMixin,pl:mixin:A}
	at net.minecraft.item.ItemStack.onItemUse(ItemStack.java:200) ~[?:?] {re:mixin,re:classloading,xf:fml:forge:filled_map.4,pl:mixin:APP:performant.mixins.json:item.ItemStackMixin,pl:mixin:APP:dankstorage.mixins.json:ItemStackMixin,pl:mixin:APP:quark.mixins.json:ItemStackMixin,pl:mixin:A}
	at net.minecraft.item.ItemStack.func_196084_a(ItemStack.java:185) ~[?:?] {re:mixin,re:classloading,xf:fml:forge:filled_map.4,pl:mixin:APP:performant.mixins.json:item.ItemStackMixin,pl:mixin:APP:dankstorage.mixins.json:ItemStackMixin,pl:mixin:APP:quark.mixins.json:ItemStackMixin,pl:mixin:A}
	at net.minecraft.client.multiplayer.PlayerController.func_217292_a(PlayerController.java:301) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:immersiveengineering.mixins.json:accessors.client.PlayerControllerAccess,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.optifine.override.PlayerControllerOF.func_217292_a(PlayerControllerOF.java:97) ~[?:?] {re:classloading}
	at net.minecraft.client.Minecraft.func_147121_ag(Minecraft.java:1366) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:randompatches.mixins.json:client.MinecraftMixin,pl:mixin:APP:immersiveengineering.mixins.json:accessors.client.MinecraftAccess,pl:mixin:APP:charm.mixins.json:accessor.MinecraftAccessor,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_184117_aA(Minecraft.java:1645) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:randompatches.mixins.json:client.MinecraftMixin,pl:mixin:APP:immersiveengineering.mixins.json:accessors.client.MinecraftAccess,pl:mixin:APP:charm.mixins.json:accessor.MinecraftAccessor,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1460) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:randompatches.mixins.json:client.MinecraftMixin,pl:mixin:APP:immersiveengineering.mixins.json:accessors.client.MinecraftAccess,pl:mixin:APP:charm.mixins.json:accessor.MinecraftAccessor,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_195542_b(Minecraft.java:953) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:randompatches.mixins.json:client.MinecraftMixin,pl:mixin:APP:immersiveengineering.mixins.json:accessors.client.MinecraftAccess,pl:mixin:APP:charm.mixins.json:accessor.MinecraftAccessor,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:607) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:randompatches.mixins.json:client.MinecraftMixin,pl:mixin:APP:immersiveengineering.mixins.json:accessors.client.MinecraftAccess,pl:mixin:APP:charm.mixins.json:accessor.MinecraftAccessor,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:184) ~[minecraft-1.16.5-client.jar:?] {re:classloading,pl:runtimedistcleaner:A}
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_281] {}
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_281] {}
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_281] {}
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_281] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider.lambda$launchService$0(FMLClientLaunchProvider.java:51) ~[forge-1.16.5-36.0.13-launcher.jar:36.0] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-8.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54) [modlauncher-8.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72) [modlauncher-8.0.9.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:82) [modlauncher-8.0.9.jar:?] {re:classloading}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:66) [modlauncher-8.0.9.jar:?] {re:classloading}
	at io.github.zekerzhayard.forgewrapper.installer.Main.main(Main.java:50) [ForgeWrapper-1.4.2.jar:?] {}
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_281] {}
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_281] {}
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_281] {}
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_281] {}
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196) [NewLaunch.jar:?] {}
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231) [NewLaunch.jar:?] {}
	at org.multimc.EntryPoint.listen(EntryPoint.java:143) [NewLaunch.jar:?] {}
	at org.multimc.EntryPoint.main(EntryPoint.java:34) [NewLaunch.jar:?] {}
```

</details>

**Comment by r-mc2** [comment]
So, I decided to do some further looking into that `ChestFrameTileEntity` class and searched this entire repo, and like you found @PrincessRTFM, nada. On a whim I opened up the search to [all of GitHub](https://github.com/search?q=ChestFrameTileEntity&type=code) and got a hit:
* [QForgeMod -- ChestFrameTileEntity](https://github.com/Qboi123/QForgeMod/blob/master/src/main/java/com/qsoftware/texturedmodels/tileentity/ChestFrameTileEntity.java)
  * File was last updated Nov. 17 -- Date of the `@version` in the file itself is Sept. 22

The odd thing is the `@author` credit in the file is PianoManu. So maybe a dependency that went missing/moved into this library? Except the author of QForgeMod, QBoi123, touts this as it's own "[random things mod](https://www.curseforge.com/minecraft/mc-mods/qforgemod)."

Some of the other base blocks and TEs of BlockCarpentry seem to exist in this repo as well. But I don't want suggest I'm implying anything negative here, probably just QBoi and Manu are working on a new mod together, sharing some code around, and Manu forgot these classes were only over on QForge's side?

**Comment by PianoManu** [comment]
Sorry for the late response, I've had a very long and rough exam phase. I tested some things and it seems, that you can't extract or insert items automatically into the chest... I'll try to fix that error. It might take some time though, since I'm pretty inexperienced in modding and I have no clue how to fix this problem...

**Comment by PianoManu** [comment]
> I'd try to track down the cause, but for some reason ChestFrameTileEntity.java isn't on the repo?
What do you mean? It's been [here](https://github.com/PianoManu/BlockCarpentry/blob/Forge-1.16.2/src/main/java/mod/pianomanu/blockcarpentry/tileentity/ChestFrameTileEntity.java) since September last year 🤷‍♂️

**Comment by PianoManu** [comment]
> The odd thing is the @author credit in the file is PianoManu. So maybe a dependency that went missing/moved into this library? Except the author of QForgeMod, QBoi123, touts this as it's own "random things mod."
> Some of the other base blocks and TEs of BlockCarpentry seem to exist in this repo as well. But I don't want suggest I'm implying anything negative here, probably just QBoi and Manu are working on a new mod together, sharing some code around, and Manu forgot these classes were only over on QForge's side?

That's interesting - no, I don't know him. I guess he took the class file (and some other files too), because he wanted to implement similar things and that's okay for me

**Comment by PianoManu** [comment]
It works now. I also tested it with botany pots like you @r-mc2 described the problem in [the related botany pots issue](https://github.com/Darkhax-Minecraft/BotanyPots/issues/138), as you can see in the video below, and it works quite fine.

https://user-images.githubusercontent.com/59123244/113565057-ac54a580-960a-11eb-9273-94a5ee396c7e.mp4

**Comment by PrincessRTFM** [comment]
> 
> 
> > I'd try to track down the cause, but for some reason ChestFrameTileEntity.java isn't on the repo?
> > What do you mean? It's been [here](https://github.com/PianoManu/BlockCarpentry/blob/Forge-1.16.2/src/main/java/mod/pianomanu/blockcarpentry/tileentity/ChestFrameTileEntity.java?rgh-link-date=2021-04-05T09%3A45%3A55Z) since September last year 🤷‍♂️

Ah! Well there you go - I was looking on the `master` branch, not the `Forge-1.16.2` branch. No wonder I couldn't find it.

**Comment by PianoManu** [comment]
Oh okay. No problem :)

**Comment by PianoManu** [comment]
I just released the fixed version [0.3.5b](https://www.curseforge.com/minecraft/mc-mods/blockcarpentry/files/3264543). The problem shouldn't occur anymore, hopefully. At least I can't recreate it anymore...

**Comment by r-mc2** [comment]
Awesome, thank you Manu! 👍
